### DISCUSSION TITLE
How to create a pull-image secrete the GitOps way

### DISCUSSION THREAD

**Discussion by haraldatbmw** [fluxcd/flux2#390] [Q&A] [answered]
I'm using a private registry with required authentication for all my application-images (frontend and backend). 
Both frontend and backend are located in kubernetes namespace "my-app".
By now I create a pull secrete with `kubectl -n my-app create secret docker-registry nexus-secret ... `.
But the application-namespace where my pull-secret must be located is created via kustomize-controller.

So if I setup a new cluster I create the cluster, apply gotk components wait for reconcile and then manually add the secrete.
That's not nice.

Is there a way to create this pull-image the "GitOps way"?

**Comment by stefanprodan** [comment, marked as answer]
Encrypt the secret and place the YAML in your repo, see https://toolkit.fluxcd.io/guides/mozilla-sops/ or https://toolkit.fluxcd.io/guides/sealed-secrets/
